ivy/functional/frontends/paddle/tensor/random.py

```diff
@@ -29,6 +29,16 @@ def poisson(x, name=None):
     return ivy.poisson(x, shape=None, device=None, dtype=None, seed=None, out=None)
 
 
+@with_supported_dtypes(
+    {"2.5.0 and below": ("float32", "float64")},
+    "paddle",
+)
```

Comment:
you should make this more version and device specific
gpu supports `float16` and 2.5.0 adds support for `bfloat16`, this decorator is only valid for 2.4.2 on cpu
I suggest using `with_supported_device_and_dtypes`